File: app/src/main/java/kyklab/quiet/QsTile.java
package kyklab.quiet;

import android.graphics.drawable.Icon;
import android.os.Build;
import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;

import androidx.annotation.RequiresApi;

@RequiresApi(api = Build.VERSION_CODES.N)
public class QsTile extends TileService {
    @Override
    public void onStartListening() {
        super.onStartListening();

        if (VolumeWatcherService.isRunning(this)) {
            setTileActive();
        }
    }

    @Override
    public void onClick() {
        super.onClick();

        if (!VolumeWatcherService.isRunning(this)) {
            startWatcherService();
            setTileActive();
        } else {
            stopWatcherService();
            setTileInactive();
        }
    }

    private void setTileActive() {
        Tile tile = getQsTile();
        tile.setIcon(Icon.createWithResource(this, R.drawable.ic_speaker_mute));
        tile.setState(Tile.STATE_ACTIVE);
        tile.updateTile();
    }

    private void setTileInactive() {
        Tile tile = getQsTile();
        tile.setIcon(Icon.createWithResource(this, R.drawable.ic_speaker));
        tile.setState(Tile.STATE_INACTIVE);
        tile.updateTile();
    }

    private void startWatcherService() {
        VolumeWatcherService.startService(this);
    }

    private void stopWatcherService() {
        VolumeWatcherService.stopService(this);
    }
}

File: app/src/main/java/kyklab/quiet/QsTile.kt
package kyklab.quiet

import android.graphics.drawable.Icon
import android.os.Build
import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import androidx.annotation.RequiresApi

@RequiresApi(api = Build.VERSION_CODES.N)
class QsTile : TileService() {
    override fun onStartListening() {
        super.onStartListening()
        if (VolumeWatcherService.isRunning(this)) {
            setTileActive()
        }
    }

    override fun onClick() {
        super.onClick()
        if (!VolumeWatcherService.isRunning(this)) {
            VolumeWatcherService.startService(this)
            setTileActive()
        } else {
            VolumeWatcherService.stopService(this)
            setTileInactive()
        }
    }

    private fun setTileActive() {
        qsTile.apply {
            icon = Icon.createWithResource(this@QsTile, R.drawable.ic_speaker_mute)
            state = Tile.STATE_ACTIVE
            updateTile()
        }
    }

    private fun setTileInactive() {
        qsTile.apply {
            icon = Icon.createWithResource(this@QsTile, R.drawable.ic_speaker)
            state = Tile.STATE_INACTIVE
            updateTile()
        }
    }
}